common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -246,17 +247,24 @@ void toAvailable() {
 
     private static final class LocalPool<T> {
         private final int ratioInterval;
-        private final Queue<DefaultHandle<T>> pooledHandles;
+        private final LocalPoolQueue<T> pooledHandles;
         private int ratioCounter;
 
+        @SuppressWarnings("unchecked")
         LocalPool(int maxCapacity, int ratioInterval, int chunkSize) {
             this.ratioInterval = ratioInterval;
-            pooledHandles = PlatformDependent.newMpscQueue(chunkSize, maxCapacity);
+            Queue<DefaultHandle<T>> queue = PlatformDependent.newMpscQueue(chunkSize, maxCapacity);
```

Comment:
The `newMpscQueue(int,int)` method considers two queue implementations, and both of them implement `MessagePassingQueue`.
I think we should be able to assert that the queue always implements this interface, and rely on it directly. No need to keep an alternative around.

Reply:
Yeah let me just cast directly but keep the old method signatures to not cause any combat issues.